opesci/staggeredgrid.py

```diff
@@ -915,12 +922,15 @@ def stress_loop(self):
                         + ccode(kernel.xreplace({self.t+1: self.time[1], self.t: self.time[0]})) + ';\n'
             dict1 = {'i': i, 'i0': i0, 'i1': i1, 'body': body}
             body = render(tmpl, dict1)
-            if self.ivdep and d == self.dimension-1:
+            if not self.pluto and self.ivdep and d == self.dimension-1:
                     body = '%s\n' % self.compiler._ivdep + body
-            if self.simd and d == self.dimension-1:
+            if not self.pluto and self.simd and d == self.dimension-1:
                     body = '#pragma simd\n' + body
 
-        if self.omp:
+        if self.pluto:
+            body = '#pragma scop\n'+body
+            body += '#pragma endscop'
+        elif self.omp:
```

Comment:
We would need to check this doesn't stop Pluto doing the right thing for the spatial loops. Inside the time loop is complicated because of couplings etc. So I'm advocating kicking into the long grass until the spatial tiling works.

On 11 Sep 2015, at 16:19, Michael Lange <notifications@github.com<mailto:notifications@github.com>> wrote:


In opesci/staggeredgrid.py<https://github.com/opesci/opesci-fd/pull/45#discussion_r39281801>:

>                      body = '#pragma simd\n' + body
>
> -        if self.omp:
> +        if self.pluto:
> +            body = '#pragma scop\n'+body
> +            body += '#pragma endscop'
> +        elif self.omp:


Is this why Pluto is only tiling the spatial loops? Should we add these in the template file around the entire time loop?

—
Reply to this email directly or view it on GitHub<https://github.com/opesci/opesci-fd/pull/45/files#r39281801>.